include/ck/tensor_operation/gpu/device/impl/device_pool3d_fwd_ndhwc_ndhwc.hpp

```diff
@@ -0,0 +1,350 @@
+
+            const index_t grid_size = (M / M_BlockTileSize);
+
+            return launch_and_time_kernel(stream_config,
+                                          kernel,
+                                          dim3(grid_size),
+                                          dim3(BlockSize),
+                                          0,
+                                          arg.a_grid_desc_m_k_,
+                                          arg.b_grid_desc_m_,
+                                          arg.in_element_op_,
+                                          arg.acc_element_op_,
+                                          float(1),
+                                          arg.p_in_dev_,
+                                          nullptr,
+                                          float(0),
+                                          arg.p_out_dev_,
+                                          arg.p_out_indices_dev_);
+        }
+
+        float Run(const BaseArgument* p_arg,
+                  const StreamConfig& stream_config = StreamConfig{}) override
+        {
+            return Run(*dynamic_cast<const Argument*>(p_arg), stream_config);
+        }
+    };
+
+    bool IsSupportedArgument(const BaseArgument* p_arg) override
+    {
+        const Argument* pArg = dynamic_cast<const Argument*>(p_arg);
+
+        if(pArg->invariant_lowest_length_ % InSrcOutDstVectorSize != 0)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    std::unique_ptr<BaseArgument>
+    MakeArgumentPointer(const void* p_in_dev,
+                        void* p_out_dev,
+                        void* p_out_indices_dev,
+                        std::array<ck::index_t, InOutRank> input_lengths,
+                        std::array<ck::index_t, WindowRank> window_lengths,
+                        std::array<ck::index_t, InOutRank> output_lengths,
+                        std::array<ck::index_t, WindowRank> window_strides,
+                        std::array<ck::index_t, WindowRank> input_left_pads,
+                        std::array<ck::index_t, WindowRank> input_right_pads,
+                        std::array<ck::index_t, WindowRank>) override
+    {
+        index_t N  = input_lengths[0];
+        index_t C  = input_lengths[1];
+        index_t Di = input_lengths[2];
+        index_t Hi = input_lengths[3];
+        index_t Wi = input_lengths[4];
+        index_t Do = output_lengths[2];
+        index_t Ho = output_lengths[3];
+        index_t Wo = output_lengths[4];
+
+        std::array<ck::index_t, WindowRank> input_spatial_lengths  = {Di, Hi, Wi};
+        std::array<ck::index_t, WindowRank> output_spatial_lengths = {Do, Ho, Wo};
+
+        return std::make_unique<Argument>(static_cast<const InDataType*>(p_in_dev),
+                                          static_cast<OutDataType*>(p_out_dev),
+                                          static_cast<int*>(p_out_indices_dev),
```

Comment:
24221716f22d23e562fb1e5638d2328cc285d2ae

Reply:
I thought it is better to extract IndexDataType to template parameter.
Otherwise, user does not know the size need to malloc to index tensor
7b83391049a7c16b67cd8c090cd2707692ae7fcd